src/movies.js

```diff
@@ -1,16 +1,93 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+
+function getAllDirectors(moviesArr) {
+  let directorsList = moviesArr.map((movie) => {
+    return movie.director;
+  });
+  return directorsList;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(arr) {
+  let dramaMovies = arr.filter((movie) => {
+    return movie.genre.includes("Drama");
+  });
+  let spielbergDramas = dramaMovies.filter((movie) => {
+    return movie.director.includes("Steven Spielberg");
+  });
+  return spielbergDramas.length;
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+let ratingsOnly = [];
+movies.forEach((r) => {
+  ratingsOnly.push(r.rate);
+});
+
+function ratesAverage(arr) {
+  if (arr.length === 0) {
+    return 0;
+  }
+  let totalRate = arr.reduce((acc, currVal) => {
+    if (currVal.rate) {
+      return acc + currVal.rate;
+    } else {
+      return acc;
+    }
+  }, 0);
+  return Number((totalRate / arr.length).toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+  let dramaMoviesArr = arr.filter((movie) => {
+    return movie.genre.includes("Drama");
+  });
+  if (dramaMoviesArr.length === 0) {
+    return 0;
+  }
+  let dramaRatesArr = dramaMoviesArr.map((r) => {
+    return r.rate;
+  });
+  let dramaRatesTotal = dramaRatesArr.reduce((acc, currVal) => {
+    return acc + currVal;
+  });
+  return Number((dramaRatesTotal / dramaRatesArr.length).toFixed(2));
+}
+
```

Comment:
You can replace lines 50-56 with `return ratesAverage(arr)`. Also I think lines 47-49 would be better placed at the top before line 44 (it makes more sense to check the length of the array first)